src/pal/src/exception/signal.cpp

```diff
@@ -554,6 +555,23 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
         ASSERT("sigprocmask failed; error is %d (%s)\n",errno, strerror(errno));
     } 
 
+#if defined(_AMD64_)
+    if (pointers->ExceptionRecord->ExceptionCode == EXCEPTION_INT_DIVIDE_BY_ZERO)
+    {
+        // It is possible that an overflow may also have been mapped to a divide-by-zero
+        // exception. This happens when we try to divide the maximum negative value of a
+        // signed integer with -1. 
+        //
+        // Thus, we will attempt to decode the instruction @ RIP to determine if that
+        // is the case using the faulting context.
+        if (IsDivByZeroAnIntegerOverflow(pointers->ContextRecord, pointers->ExceptionRecord))
```

Comment:
The difference is in diagnosability. Anytime a problem with a root cause somewhere else ends up crashing in our code, it ends up to be support cost for us.

I agree that the result will be process tear down, unless somebody registers chained signal handler to take care of their exceptions - once Mike's makes the PAL signal registration SxS friendly.